### ISSUE TITLE
Suggestion: add option for Automatic Path to Clue Location

### ISSUE THREAD

**Issue by Nick2bad4u** [Skretzo/shortest-path#39]
Is it possible for the shortest path to automatically create a path to the clue location?

**Comment by adhansen** [comment]
+1
I know there has been some discussion about integration with other plugins, but since the Clue Scroll helper plugin is part of core and this is a hub plugin it is unlikely that it will be possible to add shortest path integration on that end. A checkbox toggle for automatically setting target on clue steps would be a really nice feature.

**Comment by Skretzo** [comment]
I had a quick look at this, but it requires changes in the core Clue Scroll helper plugin to be able to know what the current clue is and where that clue location is (unless I recreate the entire clue scroll helper plugin in the shortest path plugin, but I don't want to do that).

**Comment by Nick2bad4u** [comment]
> I had a quick look at this, but it requires changes in the core Clue Scroll helper plugin to be able to know what the current clue is and where that clue location is (unless I recreate the entire clue scroll helper plugin in the shortest path plugin, but I don't want to do that).

Would it be possible to just detect the Clue scrolls by item ID in pull coordinates from the wiki? 

https://oldschool.runescape.wiki/w/Treasure_Trails/Guide/Coordinates

**Comment by Skretzo** [comment]
> Would it be possible to just detect the Clue scrolls by item ID and pull coordinates from the wiki?
> 
> https://oldschool.runescape.wiki/w/Treasure_Trails/Guide/Coordinates

Yes, but that also means I need to update that list of item IDs and coordinates every time a new clue scroll is added, which is duplicating work that the core Clue Scroll plugin will have to do anyways.

**Comment by Skretzo** [comment]
> I'm not sure if you have a way to use external coordinates to place the marker though

No, I would not be allowed to read the wiki page live if that is what you mean. The wiki page can easily be changed to contain entirely different info which most likely would break the plugin. I would have scrape that wiki page and add it to a file in my plugin, or manually type in the coordinates in a file in my plugin

**Comment by Nick2bad4u** [comment]
> > I'm not sure if you have a way to use external coordinates to place the marker though
> 
> No, I would not be allowed to read the wiki page live if that is what you mean. The wiki page can easily be changed to contain entirely different info which most likely would break the plugin. I would have to scrape that wiki page and add it to a file in my plugin, or manually type in the coordinates in a file in my plugin

I know the wiki has an API, but you could fetch it into a file in JSON format. Since the clues aren't likely to change unless there's an update to add one, I don't see why you even need to read the wiki page from the plugin directly. 

I've done a lot of work with the wiki if you want me to craft you API URL that spits out the coordinates in a list.

**Comment by Skretzo** [comment]
> I've done a lot of work with the wiki if you want me to craft you API URL that spits out the coordinates in a list.

I don't think RuneLite or the wiki wants us to do that, and it should instead be a static file. But keeping and updating such a file is not really something I want to do so I will see if core RuneLite accepts the necessary changes to expose the coordinate of the current active clue scroll. If not then I don't think I will bother adding this feature, sadly.

**Comment by Nick2bad4u** [comment]
> > I've done a lot of work with the wiki if you want me to craft you API URL that spits out the coordinates in a list.
> 
> I don't think RuneLite or the wiki wants us to do that, and it should instead be a static file. But keeping and updating such a file is not really something I want to do so I will see if core RuneLite accepts the necessary changes to expose the coordinate of the current active clue scroll. If not then I don't think I will bother adding this feature, sadly.

No worries. I want the API use as a developer tool to easily pull the lists without typing them, not for the direct plugin to pull the API cords 

I get it though it's a lot of work for a simple qol fix. Thanks for looking into it

**Comment by Infinitay** [comment]
> I had a quick look at this, but it requires changes in the core Clue Scroll helper plugin to be able to know what the current clue is and where that clue location is (unless I recreate the entire clue scroll helper plugin in the shortest path plugin, but I don't want to do that).

Perhaps it's better suited to add support for [Zoink's Clue Details plugin](https://github.com/Zoinkwiz/clue-details/). I had the idea of hooking Shortest Path to clues before after coming across Zoink's plugin, but I never found the time and forgot about it.

**Comment by Skretzo** [comment]
> > I had a quick look at this, but it requires changes in the core Clue Scroll helper plugin to be able to know what the current clue is and where that clue location is (unless I recreate the entire clue scroll helper plugin in the shortest path plugin, but I don't want to do that).
> 
> Perhaps it's better suited to add support for [Zoink's Clue Details plugin](https://github.com/Zoinkwiz/clue-details/). I had the idea of hooking Shortest Path to clues before after coming across Zoink's plugin, but I never found the time and forgot about it.

That is a good idea, as long as Zoink's Clue Details plugin can do everything the core Clue Scroll plugin can do or the two plugins can be used seamlessly together. One of the big advantages of having the a clue scroll plugin decide when the show the path is that it probably knows when a new clue has been received, which is something the shortest path plugin has no idea of.
